slides/r-adoption-jan2023.qmd

```diff
@@ -0,0 +1,204 @@
+
+-   Prepare public training materials to disseminate best practice for software engineering in the Biostatistics community
+    -   At the beginning of February, a face-to-face workshop will take place in Basel, Switzerland with a focus on open-source software for clinical trials
+    -   Organize conference sessions with a focus on statistical software engineering at CEN, JSM and ASA/FDA Workshop
+    -   Video clips in preparation with a focus on best practices for software engineering
+
+## Why it's not just another package
+
+-   Ongoing maintenance and support from the pharmaceutical industry that is supported by American Statistical Association
+-   Package is part of the mission, but to emphasize our goal is to push out information on practices for engineering high-quality open-source statistical software
+
+## Long term perspective
+
+-   Software engineering is a critical competence in producing high-quality statistical software
+-   A lot of work need to be done regarding the establishment, dissemination and adoption of best practices for engineering open-source software
+-   Improving the way software engineering is done will help improve the efficiency, reliability and innovation within Biostatistics
+
+## Comparing SAS and R
+
+- Estimation Methods
+- Covariance structures
+- Degrees of Freedom
+- contrasts
+- comparing estimates
+- run times
+
+## Estimation Methods
+
+|       Method     | mmrm::mmrm | SAS |
+|:----------------|:----------:|:---:|
+|       ML         |     X      |  X  |
+|       REML       |     X      |  X  |
+|       MIVQUE0*   |            |  X  |
+
+*MIVQUE0: minimum variance quadratic unbiased estimation of the covariance parameters
+
+##  Covariance structures
+
+- SAS has 23 non-spatial covariance structures, while mmrm has 10. 
+  - 9 structures intersect with SAS
+  - Ante-dependence (homogeneous) is only in the mmrm package. 
+- SAS has 14 spatial covariance structures compared to the spatial exponential one available in mmrm.
+
+##  Covariance structures details
+
+|       Covariance structures                   | mmrm::mmrm   |  SAS  |
+|:----------------------------------------------|:------------:|:-----:|
+|     Unstructured (Unweighted/Weighted)        |     X/X      |  X/X  |
+|     Toeplitz (hetero/homo)      |     X/X      |  X/X  |
+| Compound symmetry (hetero/homo) |     X/X      |  X/X  |
+|  Auto-regressive (hetero/homo)  |     X/X      |  X/X  |
+|  Ante-dependence (hetero/homo)  |     X/X      |  X    |
+|        Spatial exponential                    |     X        |  X    |
+
+## Degrees of Freedom Methods
+
+|       Method     | mmrm::mmrm | SAS |
+|:----------------|:----------:|:---:|
+|       Contain         |      X*     |  X  |
+|       Between/Within  |       X* |      X  |
+|       Residual         |    X*       |  X  |
+|       Satterthwaite         |     X      |  X  |
+|       Kenward-Roger       |     X      |  X  |
+|       Kenward-Roger (Linear)   |  X          |  X  |
+
```

Comment:
to be fair, could mention that the second Kenward-Roger KR2 method is not implemented in mmrm for now.

Reply:
I thought KR2 was Kenward-Roger (Linear) 

from [SAS docs](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_mixed_syntax10.htm#statug.mixed.modelstmt_ddfm)

The DDFM=KENWARDROGER2 option specifies an improved approximation of the DDFM=KENWARDROGER method that uses a less biased precision estimator, as proposed by Kenward and Roger ([2009](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_mixed_references.htm#statug_mixedkenw_m09)). For an intrinsically linear covariance parameterization, this option produces the same precision estimator as that obtained using DDFM=KR(FIRSTORDER).

Reply:
No it is the last one from this SAS excerpt

Reply:
From what I understand `KR(FIRSTORDER) == KR2`

Reply:
I think we have integration tests comparing with that, so I am pretty sure it is not KR2. Actually James Roger emailed us on this topic. Agree that from SAS text it is confusing.